enhancements/installer/vsphere-ipi-zonal.md

```diff
@@ -0,0 +1,1067 @@
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The goal of this enhancement is to provide the ability to install in a
+vSphere environment with multiple datacenters and clusters.
+
+This will be an opinionated design, the vCenter datacenter will always be a `region`
+and a vCenter cluster will always be a `zone`.
+
+## Motivation
+
+Users of OpenShift would like the ability to deploy
+within multiple datacenters and clusters to increase
+reliability. Customers would also like to take advantage
+of the concept of regions and zoning that this type of deployment would
+offer.
+
+- https://issues.redhat.com/browse/RFE-845
+- https://issues.redhat.com/browse/OCPPLAN-4927
+
+### Goals
+
+- Support regions and zones in vSphere using multiple datacenters (region) and
+clusters (zone)
+- Support installation into multiple datacenters and multiple clusters
+
+### Non-Goals
+
+- Support multiple subnets
+- Support multiple vcenters
+
+Note: The platform spec will be modified to support this at a future date.
+Only a single item in `vcenters` will be supported for the initial release.
+
+## Proposal
+
+Modification of the installer to support the provisioning of masters in defined
+datacenters and clusters.
+
+### Workflow Description
+
+### api
+
+#### Infrastructure
+
+Using the upstream [vSphere cluster api](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/apis/v1beta1/vspherefailuredomain_types.go)
+and the upstream [vSphere cloud config manager](https://github.com/kubernetes/cloud-provider-vsphere/blob/master/pkg/common/config/types_yaml.go)
+as examples to implement parameters of `VSpherePlatformSpec`.
+These parameters include the optional and required information to manage a OpenShift cluster on vSphere.
+
+```golang
+// VSpherePlatformFailureDomainType is the name of the failure domain type.
+// There are two defined failure domains currently, Datacenter and ComputeCluster.
+// Each represents a vCenter object type within a vSphere environment.
+// +kubebuilder:validation:Enum=HostGroup;Datacenter;ComputeCluster
+// + ---
+// + HostGroup is included for future support, consumers should output an error
+// + if this value is set.
+type VSpherePlatformFailureDomainType string
+
+const (
+	// HostGroupFailureDomain as a type allows the use of a group of ESXi hosts
+	// to be represented as a failure domain zone. When using this
+	// case it is expected that region would be a cluster.
+	// HostGroups within vCenter must be preconfigured and
+	// assigned in the topology.
+	HostGroupFailureDomain VSpherePlatformFailureDomainType = "HostGroup"
+
+	// ComputeClusterFailureDomain failure domain can either be a zone or region.
+	// The vCenter cluster is required to preconfigured and
+	// assigned in the topology.
+	ComputeClusterFailureDomain VSpherePlatformFailureDomainType = "ComputeCluster"
```

Comment:
What is a zone if this is a region? Can we have zones within a ComputeCluster?

Reply:
Removed with api changes